<?xml version="1.0" ?>
<launch>

  <arg name="camera_name0"	default="live_camera"/>
  <arg name="image_topic0"	default="image_raw"/>
  <arg name="camera_name1"	default="realsense"/>
  <arg name="image_topic1"	default="color/image_raw"/>

  <arg name="detector_name"	default="aruco_multi_detector"/>
  <arg name="camera_names"	default="$(arg camera_name0)
					 $(arg camera_name1)"/>
  <arg name="image_topics"	default="$(arg image_topic0)
					 $(arg image_topic1)"/>
  <arg name="marker_map_dir"	default="$(dirname)/../config"/>
  <arg name="marker_map"	default="calib_board-6x4"/>
  <arg name="marker_frame"	default="marker_frame"/>
  <arg name="reference_frame"	default=""/>
  <arg name="manager"		default=""/>

  <node unless="$(eval manager == '')"
	name="$(arg manager)" pkg ="nodelet" type="nodelet"
	args="manager" output="screen">
    <param name="num_worker_threads"	value="4"/>
  </node>

  <node name="$(arg camera_name0)"
  	pkg="usb_cam" type="usb_cam_node" output="screen">
    <rosparam file="$(find aist_routines)/config/$(arg camera_name0).yaml"/>
  </node>

  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="camera"			value="$(arg camera_name1)"/>
    <arg name="tf_prefix"		value="$(arg camera_name1)"/>
    <arg name="enable_depth"		value="true"/>
    <arg name="enable_infra1"		value="false"/>
    <arg name="enable_infra2"		value="false"/>
    <arg name="enable_pointcloud"	value="false"/>
    <arg name="align_depth"		value="true"/>
    <arg name="external_manager"	value="$(eval manager != '')"/>
    <arg name="manager"			value="/$(arg manager)"
	 if="$(eval manager != '')"/>
  </include>

  <include file="$(dirname)/run_multi.launch">
    <arg name="detector_name"	value="$(arg detector_name)"/>
    <arg name="camera_names"	value="$(arg camera_names)"/>
    <arg name="image_topics"	value="$(arg image_topics)"/>
    <arg name="marker_map_dir"	value="$(arg marker_map_dir)"/>
    <arg name="marker_map"	value="$(arg marker_map)"/>
    <arg name="marker_frame"	value="$(arg marker_frame)"/>
    <arg name="reference_frame"	value="$(arg reference_frame)"/>
    <arg name="manager"		value="$(arg manager)"/>
  </include>

  <node name="$(anon rviz)" pkg="rviz" type="rviz" output="screen"
	args="-d $(dirname)/multicam.rviz"/>

  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>

</launch>
